core/ecschema-editing/src/Merging/EnumerationMerger.ts

```diff
@@ -39,7 +39,7 @@ export async function addEnumeration(context: SchemaMergeContext, change: Enumer
 export async function modifyEnumeration(context: SchemaMergeContext, change: EnumerationDifference, itemKey: SchemaItemKey) {
   const enumeration = await context.targetSchema.lookupItem(itemKey) as MutableEnumeration;
   if(change.difference.type !== undefined) {
-    throw new Error(`The Enumeration ${itemKey.name} has an incompatible type. It must be "${primitiveTypeToString(enumeration.type!)}", not "${change.difference.type}".`);
+    throw new Error(`The Enumeration ${itemKey.name} has an incompatible type. It must be "${enumeration.type ? primitiveTypeToString(enumeration.type) : "<unknown>"}", not "${change.difference.type}".`);
```

Comment:
Why is the underlying Enumeration._type property possibly undefined?  Any valid enumeration MUST have a type set so we should not force callers to test if the type is null.

Reply:
@ColinKerr The constructor also has this as optional. It's internal so we can change this. Should we require the parameter or have a default (int or string)?